@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1668,12 +1661,14 @@ export class TextElementsRenderer {
                 if (placementStats) {
                     placementStats.numPoiTextsInvisible++;
                 }
-                labelState.reset();
-                return false;
+                if (!renderIcon || iconInvisible) {
+                    labelState.reset();
+                    return false;
+                }
```

Comment:
with this change, an invisible text without icon will be faded in later, since textNeedsDraw is evaluated true and textRejected is false.

Reply:
fixed it by adding the reset for the textRenderState and by handling the invisible text label in the code below (not rendering it).